include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_dual_splitting.cpp

```diff
@@ -712,6 +712,24 @@ TimeIntBDFDualSplitting<dim, Number>::viscous_step()
       velocity_np = velocity_viscous_last_iter;
     }
 
+    /*
+     *  update variable viscosity
+     */
+    if(this->param.viscous_problem() and this->param.viscosity_is_variable() and
+       this->param.treatment_of_variable_viscosity == TreatmentOfVariableViscosity::Explicit)
+    {
+      dealii::Timer timer_viscosity_update;
+      timer_viscosity_update.restart();
+
+      pde_operator->update_viscosity(velocity_np);
+
+      if(this->print_solver_info() and not(this->is_test))
+      {
+        this->pcout << std::endl << "Update of variable viscosity:";
+        print_wall_time(this->pcout, timer_viscosity_update.wall_time());
+      }
+    }
+
```

Comment:
just to double-check:
in case we execute the above block (viscosity update),
the first branch below is **not** taken since `nonlinear_problem_has_to_be_solved()` is false given the explicit treatment.
So actually these changes now just pull out the update, adding another condition.
You think it is easier to read this way, right? The actual behavior is the same still, is it (if the `viscous_problem()` were added in the old version)?